Sharing link to post crashes the app

**Describe the bug**

Whenever I try using the "share link to post" the app crashes.

**To Reproduce**
Steps to reproduce the behavior:
1. Open the app
2. Click on the 3 dots button to the bottom right of any post
3. Select "Share link to post"
4. The app crashes

**Expected behavior**

The app correctly shows the share interface

**Screenshots or video**

https://github.com/pachli/pachli-android/assets/1263058/94337ba9-d6b6-4050-b284-f8704137b053

**Versions**

Pachli 2.5.1
Google Pixel 3a
Android 12
SDK level 32

**Additional context**

Sometimes the following popup appears when the app crashes

![image](https://github.com/pachli/pachli-android/assets/1263058/bb48d5c8-c34f-4515-92ed-79fc5df4127b)

**Affirmation**
I have checked other issues for this project (open and closed) and I cannot find one that matches the problem I am reporting.

I couldn't get the crash report via the error report but apparently I can see it in LogCat!

```
2024-05-17 16:12:50.356  6057-6057  AndroidRuntime          android                              E  FATAL EXCEPTION: main
Process: android:ui, PID: 6057
java.lang.RuntimeException: Unable to start activity ComponentInfo{android/com.android.internal.app.ChooserActivity}: java.lang.RuntimeException: Parcelable encountered ClassNotFoundException reading a Serializable object (name = app.pachli.core.activity.extensions.TransitionKind)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3707)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3864)
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loopOnce(Looper.java:201)
at android.os.Looper.loop(Looper.java:288)
at android.app.ActivityThread.main(ActivityThread.java:7870)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
Caused by: java.lang.RuntimeException: Parcelable encountered ClassNotFoundException reading a Serializable object (name = app.pachli.core.activity.extensions.TransitionKind)
at android.os.Parcel.readSerializable(Parcel.java:3513)
at android.os.Parcel.readValue(Parcel.java:3277)
at android.os.Parcel.readArrayMapInternal(Parcel.java:3623)
at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
at android.os.BaseBundle.unparcel(BaseBundle.java:236)
at android.os.Bundle.getParcelable(Bundle.java:1002)
at android.content.Intent.getParcelableExtra(Intent.java:8548)
at com.android.internal.app.ChooserActivity.onCreate(ChooserActivity.java:607)
at android.app.Activity.performCreate(Activity.java:8054)
at android.app.Activity.performCreate(Activity.java:8034)
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1341)
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3688)
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3864) 
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loopOnce(Looper.java:201) 
at android.os.Looper.loop(Looper.java:288) 
at android.app.ActivityThread.main(ActivityThread.java:7870) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
Caused by: java.lang.ClassNotFoundException: app.pachli.core.activity.extensions.TransitionKind
at java.lang.Class.classForName(Native Method)
at java.lang.Class.forName(Class.java:536)
at android.os.Parcel$2.resolveClass(Parcel.java:3499)
at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1733)
at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1624)
at java.io.ObjectInputStream.readEnum(ObjectInputStream.java:1854)
at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1439)
at java.io.ObjectInputStream.readObject(ObjectInputStream.java:430)
at android.os.Parcel.readSerializable(Parcel.java:3507)
at android.os.Parcel.readValue(Parcel.java:3277) 
at android.os.Parcel.readArrayMapInternal(Parcel.java:3623) 
at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292) 
at android.os.BaseBundle.unparcel(BaseBundle.java:236) 
at android.os.Bundle.getParcelable(Bundle.java:1002) 
at android.content.Intent.getParcelableExtra(Intent.java:8548) 
at com.android.internal.app.ChooserActivity.onCreate(ChooserActivity.java:607) 
at android.app.Activity.performCreate(Activity.java:8054) 
at android.app.Activity.performCreate(Activity.java:8034) 
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1341) 
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3688) 
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3864) 
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loopOnce(Looper.java:201) 
at android.os.Looper.loop(Looper.java:288) 
at android.app.ActivityThread.main(ActivityThread.java:7870) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
Caused by: java.lang.ClassNotFoundException: app.pachli.core.activity.extensions.TransitionKind
at java.lang.Class.classForName(Native Method) 
at java.lang.Class.forName(Class.java:536) 
at android.os.Parcel$2.resolveClass(Parcel.java:3499) 
at java.io.ObjectInputStream.readNonProxyDesc(ObjectInputStream.java:1733) 
at java.io.ObjectInputStream.readClassDesc(ObjectInputStream.java:1624) 
at java.io.ObjectInputStream.readEnum(ObjectInputStream.java:1854) 
at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1439) 
at java.io.ObjectInputStream.readObject(ObjectInputStream.java:430) 
at android.os.Parcel.readSerializable(Parcel.java:3507) 
at android.os.Parcel.readValue(Parcel.java:3277) 
at android.os.Parcel.readArrayMapInternal(Parcel.java:3623) 
at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292) 
at android.os.BaseBundle.unparcel(BaseBundle.java:236) 
at android.os.Bundle.getParcelable(Bundle.java:1002) 
at android.content.Intent.getParcelableExtra(Intent.java:8548) 
at com.android.internal.app.ChooserActivity.onCreate(ChooserActivity.java:607) 
at android.app.Activity.performCreate(Activity.java:8054) 
at android.app.Activity.performCreate(Activity.java:8034) 
at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1341) 
at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3688) 
at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3864) 
at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253) 
at android.os.Handler.dispatchMessage(Handler.java:106) 
at android.os.Looper.loopOnce(Looper.java:201) 
at android.os.Looper.loop(Looper.java:288) 
at android.app.ActivityThread.main(ActivityThread.java:7870) 
at java.lang.reflect.Method.invoke(Native Method) 
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)  
```

--- a/app/src/main/java/app/pachli/fragment/SFragment.kt
+++ b/app/src/main/java/app/pachli/fragment/SFragment.kt
@@ -95,7 +95,11 @@ abstract class SFragment<T : IStatusViewData> : Fragment(), StatusActionListener
     private var serverCanTranslate = false
 
     override fun startActivity(intent: Intent) {
-        requireActivity().startActivityWithDefaultTransition(intent)
+        if (intent.component?.className?.startsWith("app.pachli.") == true) {
+            requireActivity().startActivityWithDefaultTransition(intent)
+        } else {
+            super.startActivity(intent)
+        }
     }
 
     override fun onAttach(context: Context) {

--- a/core/activity/src/main/kotlin/app/pachli/core/activity/extensions/ActivityExtensions.kt
+++ b/core/activity/src/main/kotlin/app/pachli/core/activity/extensions/ActivityExtensions.kt
@@ -27,8 +27,12 @@ import app.pachli.core.activity.BaseActivity
 import app.pachli.core.activity.BuildConfig
 
 /**
- * Starts the activity in [intent] (which must be a subclass of [BaseActivity])
- * using [transitionKind] as the open/close transition.
+ * Starts the activity in [intent]. Should only be called on subclasses of [BaseActivity].
+ *
+ * If the activity is a Pachli activity then [transitionKind] is included in the intent and
+ * used as the open/close transition.
+ *
+ * See [BaseActivity.onCreate] for the other half of this code.
  */
 fun Activity.startActivityWithTransition(intent: Intent, transitionKind: TransitionKind) {
     if (BuildConfig.DEBUG) {
@@ -37,7 +41,9 @@ fun Activity.startActivityWithTransition(intent: Intent, transitionKind: Transit
         }
     }
 
-    intent.putExtra(EXTRA_TRANSITION_KIND, transitionKind)
+    if (intent.component?.className?.startsWith("app.pachli.") == true) {
+        intent.putExtra(EXTRA_TRANSITION_KIND, transitionKind)
+    }
     startActivity(intent)
 
     if (canOverrideActivityTransitions()) {
@@ -60,7 +66,7 @@ fun Activity.startActivityWithDefaultTransition(intent: Intent) = startActivityW
 fun Activity.setCloseTransition(transitionKind: TransitionKind) {
     if (BuildConfig.DEBUG) {
         if (this !is BaseActivity) {
-            throw IllegalStateException("startActivityWithTransition must be used with BaseActivity subclass")
+            throw IllegalStateException("setCloseTransition must be used with BaseActivity subclass")
         }
     }
 

--- a/core/activity/src/main/kotlin/app/pachli/core/activity/extensions/ActivityExtensions.kt
+++ b/core/activity/src/main/kotlin/app/pachli/core/activity/extensions/ActivityExtensions.kt
@@ -40,8 +40,8 @@ fun Activity.startActivityWithTransition(intent: Intent, transitionKind: Transit
             throw IllegalStateException("startActivityWithTransition must be used with BaseActivity subclass")
         }
     }
-    intent.putExtra(EXTRA_TRANSITION_KIND, transitionKind)
 
+    intent.putExtra(EXTRA_TRANSITION_KIND, transitionKind)
     startActivity(intent)
 
     if (canOverrideActivityTransitions()) {

--- a/core/activity/src/main/kotlin/app/pachli/core/activity/extensions/ActivityExtensions.kt
+++ b/core/activity/src/main/kotlin/app/pachli/core/activity/extensions/ActivityExtensions.kt
@@ -40,10 +40,8 @@ fun Activity.startActivityWithTransition(intent: Intent, transitionKind: Transit
             throw IllegalStateException("startActivityWithTransition must be used with BaseActivity subclass")
         }
     }
+    intent.putExtra(EXTRA_TRANSITION_KIND, transitionKind)
 
-    if (intent.component?.className?.startsWith("app.pachli.") == true) {
-        intent.putExtra(EXTRA_TRANSITION_KIND, transitionKind)
-    }
     startActivity(intent)
 
     if (canOverrideActivityTransitions()) {

